pysingfel/experiment/base.py

```diff
@@ -0,0 +1,107 @@
+import numpy as np
+import pysingfel.geometry as psg
+from pysingfel.reciprocal_detector import ReciprocalDetector
+
+
+class Experiment(object):
+    mesh_size = 151
+
+    def __init__(self, det, beam, particles):
+        self.det = det
+        self.beam = beam
+        self.n_particle_kinds = len(particles)
+
+        # Create mesh
+        highest_k_beam = self.beam.get_highest_wavenumber_beam()
+        recidet = ReciprocalDetector(self.det, highest_k_beam)
+        mesh, self.voxel_length = recidet.get_reciprocal_mesh(
+            voxel_number_1d=self.mesh_size)
+
+        # Create volumes
+        import pysingfel.gpu as pg
+        self.volumes = []
+        for particle in particles:
+            if particle is None:
+                self.volumes.append(np.zeros(mesh.shape[:-1], np.complex128))
+                continue
+            self.volumes.append(
+                pg.calculate_diffraction_pattern_gpu(mesh, particle, return_type='complex_field'))
+
+    def generate_image(self):
+        img_stack = self.generate_image_stack()
+        return self.det.assemble_image_stack(img_stack)
+
+    def generate_image_stack(self, return_photons=None,
```

Comment:
set return_photons=True
change logic for return_photons in this function

Reply:
That behavior was intended. The idea was that `generate_image_stack` would return what was asked. If nothing is asked, then the photons. `generate_image_stack(return_intensities=True)` was meant to only return the intensities.